### PR TITLE
fix(tooltip): implement change detection

### PR DESCRIPTION
**Pull request by peterblazejewicz**
This fixes issues with custom tooltip class changes  not being applied after
tooltip creation.

Fixes: #3335

### CODE REVIEW

**Comment by joergplewe** [comment]
Great news. Will it be backported to the v4 release of ng-bootstrap? (I'm stuck with Angular7)

File: src/tooltip/tooltip.ts [hunk 1]
```diff
@@ -253,6 +264,13 @@ export class NgbTooltip implements OnInit, OnDestroy {
         this.close.bind(this), +this.openDelay, +this.closeDelay);
   }
 
+  ngOnChanges(changes: SimpleChanges) {
```
Comment (maxokorokov) [inline src/tooltip/tooltip.ts]: just a shorter variant:

```ts
ngOnChanges({tooltipClass}: SimpleChanges) { }
```

File: src/tooltip/tooltip.spec.ts [hunk 2]
```diff
@@ -60,6 +61,78 @@ describe('ngb-tooltip-window', () => {
...
+    fixture.detectChanges();
+    fixture.componentInstance.tooltipClass = 'second-custom-class';
+    fixture.detectChanges();
+    fixture.componentInstance.tooltipClass = 'third-custom-class';
+    fixture.detectChanges();
+
+    expect(fixture.nativeElement).toHaveCssClass('third-custom-class');
+  });
+
+  describe('ngb-tooltip-window tooltipClass change detection', () => {
```
Comment (maxokorokov) [inline src/tooltip/tooltip.spec.ts]: Check my other comment, it's Angular who will call change detection if any of the inputs change regardless of what you do. So this `describe` looks useless to me.

File: src/tooltip/tooltip.spec.ts [hunk 3]
```diff
@@ -60,6 +61,78 @@ describe('ngb-tooltip-window', () => {
     expect(fixture.nativeElement).toHaveCssClass('my-custom-class');
   });
 
+  it('should allow to change custom class to new one', () => {
```
Comment (maxokorokov) [inline src/tooltip/tooltip.spec.ts]: This is kinda covered by the previous test in this file...

What you should actually test is that:
- when you change `classThatChanges` → `<div [ngbTooltip]="hi" [tooltipClass]="classThatChanges">`
- it actually changes class on the `NgbTooltipWindow`

File: src/tooltip/tooltip.ts (hunk 1 above)
Comment (peterblazejewicz) [reply src/tooltip/tooltip.ts]: redacted

File: src/tooltip/tooltip.spec.ts (hunk 3 above)
Comment (peterblazejewicz) [reply src/tooltip/tooltip.spec.ts]: redacted, thanks!

File: src/tooltip/tooltip.spec.ts (hunk 2 above)
Comment (peterblazejewicz) [reply src/tooltip/tooltip.spec.ts]: redacted, thanks!

File: src/tooltip/tooltip.ts [hunk 4]
```diff
@@ -39,16 +41,25 @@ let nextId = 0;
...
   styleUrls: ['./tooltip.scss']
 })
-export class NgbTooltipWindow {
+export class NgbTooltipWindow implements OnChanges {
   @Input() id: string;
   @Input() tooltipClass: string;
+
+  constructor(private _changeDetector: ChangeDetectorRef) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
```
[the remark this answers is not in the document]
Comment (peterblazejewicz) [reply src/tooltip/tooltip.ts]: redacted, thanks!

**Comment by peterblazejewicz** [comment]
reworked, updated. Thanks!

**Pull request merged**